src/interfaces/collections/handlers/collections-query.ts

```diff
@@ -153,4 +159,42 @@ async function fetchUnpublishedRecordsByRequester(collectionsQuery: CollectionsQ
 
   const unpublishedRecordsForRequester = await messageStore.query(includeCriteria, excludeCriteria);
   return unpublishedRecordsForRequester;
-}
\ No newline at end of file
+}
+
+/**
+ * Handles dateSort field passed in the CollectionsQuery. There are 4 options for dateSort:
+ * 1. createdAscending - Sort in ascending order based on when the message was created
+ * 2. createdDescending - Sort in descending order based on when the message was created
+ * 3. publishedAscending - If the message is published, sort in asc based on publish date
+ * 4. publishedDescending - If the message is published, sort in desc based on publish date
+ *
+ * If dateSort is not present we return the unmodified list of entries. This method throws
+ * if there is an unsupported DateSort operation.
+ *
+ * @param collectionsQuery - Underlying Collections Query
+ * @param entries - Entries to be sorted if dateSort is present
+ * @returns List of Messages
+ */
+async function handleDateSort(
+  collectionsQuery: CollectionsQuery,
+  entries: BaseMessage[]
+): Promise<BaseMessage[]> {
+
+  const { dateSort } = collectionsQuery.message.descriptor;
+  const collectionMessages = entries as CollectionsWriteMessage[];
+
+  switch (dateSort) {
+  case DateSort.CreatedAscending:
+    return collectionMessages.sort((a, b) => lexicographicalCompare(a.descriptor.dateCreated, b.descriptor.dateCreated));
```

Comment:
are `dateCreated` supposed to be a timestamp in string format?  if so, would it be easier/better to `parseInt` and compare numerically instead?

Reply:
`dateCreated` is in ISO 8061 format, not unix epoch time. e.g.

`2022-12-21T10:20:30.405060`

Not sure what `{ numeric: true }` will do with `localCompare` but don't think we can use `parseInt`.